skrub/_fuzzy_join.py

```diff
@@ -400,39 +436,57 @@ def fuzzy_join(
     main_num_cols = main_table[main_cols].select_dtypes(include="number").columns
     aux_num_cols = aux_table[aux_cols].select_dtypes(include="number").columns
 
+    main_time_cols = main_table[main_cols].select_dtypes(include="datetime").columns
+    aux_time_cols = aux_table[aux_cols].select_dtypes(include="datetime").columns
+
+    main_str_cols = list(set(main_cols) - set(main_num_cols) - set(main_time_cols))
```

Comment:
IMO we shouldn't assume that the remaining columns are strings (for instance, boolean?) and explicitly query the string or object datatype to avoid surprises.